src/Plugin.php

```diff
@@ -264,4 +277,30 @@ public static function renderTemplate(array $template_subpathnames, array $varia
     }
   }
 
+  /**
+   * Cron event callback to ensure proper database indices.
+   */
+  public static function cron_ensure_indices() {
```

Comment:
As there will be a few more indexes, I'd recommend to move this code here (without the triggering code in init) either into a new Database class or the existing Schema class.